saleor/graphql/order/filters.py

```diff
@@ -79,12 +80,23 @@ def filter_status(qs, _, value):
 
 
 def filter_customer(qs, _, value):
-    qs = qs.filter(
-        Q(user_email__ilike=value)
-        | Q(user__email__trigram_similar=value)
-        | Q(user__first_name__trigram_similar=value)
-        | Q(user__last_name__trigram_similar=value)
-    )
+    try:
+        validate_email(value)
+        qs = qs.filter(Q(user_email__iexact=value) | Q(user__email__iexact=value))
+    except ValidationError:
+        try:
+            first_name, last_name = value.split()
+            qs = qs.filter(
+                user__first_name__iexact=first_name, user__last_name__iexact=last_name
+            )
```

Comment:
Is this using a DB index?

Reply:
No, there is no index on `first_name` or `last_name` fields. Would you suggest adding those indexes or utilize `User.search_document` for this search?

Reply:
Per conversation, indexes will be added for `first_name` and `last_name`.